### DISCUSSION TITLE
Best Practices Portable K3s Cluster

### DISCUSSION THREAD

**Discussion by oheckmann** [k3s-io/k3s#6717] [Q&A]
I am trying to set up a K3s cluster on a RPI cluster in such a way that it can be transferred to areas with different LANs and still maintain functionality, in a way that is "Plug and Play".  What changes do I need to do to make to the K3s installations to best enable this?

**Comment by brandond** [comment]
Kubernetes generally expects that the node IPs will not change once they have joined the cluster. If you are going to set up a "portable" cluster, I would ensure that the nodes all run on a stable local network that moves with them, where their addresses will not change even as you move from site to site.

**Reply by oheckmann** [reply]
The Cluster is connected together with Ethernet cables, is it possible to configure the nodes in the K3s cluster to communicate with each other using Ethernet instead of WiFi?

**Reply by brandond** [reply]
have you tried setting `--node-ip`?

**Reply by oheckmann** [reply]
To make sure that I am understanding the documentation correctly: for each node in the cluster, you reinstall k3s with the --node-ip and the ip address that the k3s node is currently using (or another one to try to avoid collisions)

**Reply by brandond** [reply]
You should do it when originally joining the nodes to the cluster; don't change the IP once they've joined (for the reasons discussed above).

**Reply by oheckmann** [reply]
Is it possible to configure a node currently joined to a cluster to set its current IP as if it was set by --node-ip?

**Reply by brandond** [reply]
A better idea is to stop k3s on each node, delete it from the cluster, then rejoin it with the correct IP.

**Reply by oheckmann** [reply]
Is there a recommended way to choose what a "correct IP" should be? Or is it more a situation where you pick one, such as the IP that the node is currently using, and fix it that way?

**Reply by brandond** [reply]
If they're connected via a private network that is used just for the cluster, then you should be able to figure out how to give each node a reserved IP address on that network, right? Use that IP.

**Reply by oheckmann** [reply]
Is it possible to configure nodes in the cluster to connect with other nodes in the cluster over Ethernet, rather than LAN?

**Reply by brandond** [reply]
yes, by using the LAN IPs instead as we've been discussing?
